astroid/inference.py

```diff
@@ -58,9 +71,24 @@
 from astroid.interpreter import dunder_lookup
 from astroid.manager import AstroidManager
 
+if TYPE_CHECKING:
+    from astroid.objects import Property
+
 # Prevents circular imports
 objects = util.lazy_import("objects")
 
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+FunctionDefT = TypeVar("FunctionDefT", bound=nodes.FunctionDef)
+
+
+class InferenceErrorTD(TypedDict):
+    node: nodes.NodeNG
+    context: Optional[InferenceContext]
```

Comment:
What about `InferenceErrorInfo`? Can you also move it to `decorators.py` or maybe a new `typing.py` file? See also: https://github.com/PyCQA/astroid/pull/1436#discussion_r821198275